src/hooks/sourceNodes/index.js

```diff
@@ -72,6 +72,51 @@ module.exports = async (
             { deserializeResponse: false, allPages: true },
           );
           if (payload) {
+            // `rich_text`, `links`, `link` fields link to other entities and we need to
+            // fetch them separately to make sure we have all the data
+            const linkedEntitiesIdsToFetch = payload.data.reduce(
+              (acc, payload) => {
+                const item_type_rel = payload.relationships.item_type.data;
+                const itemTypeForThis = loader.entitiesRepo.findEntity(
+                  item_type_rel.type,
+                  item_type_rel.id,
+                );
+                const fieldsToResolve = itemTypeForThis.fields.filter(
+                  fieldDef =>
+                    [`rich_text`, `links`, `link`].includes(fieldDef.fieldType),
+                );
+
+                const fieldsToResolveKeys = fieldsToResolve.map(
+                  field => field.apiKey,
+                );
+                fieldsToResolveKeys.forEach(f => {
+                  const v = payload.attributes[f];
```

Comment:
@pieh this looks fine, but fields can be localized:

```js
const localizedFieldsToResolve = fieldsToResolve.filter(field => f.localized);
```

 In this case `payload.attributes[f]` would be something like this (suppose it's a Links/Rich text field):

```
{
  'en': ['123', '456'],
  'it': ['789'],
}
```

so you would have to additionally iterate over the object keys to get all the IDs for all the locales.

Reply:
Thanks for this information. I will set up translatable model in my test space to account for this case and will update the PR to cover it

Reply:
Pushed https://github.com/datocms/gatsby-source-datocms/pull/122/commits/e7930ffc5815c0c76b64e0c0f21f812980c24f1a to handle localized fields case (and while I was at it also used some more reasonable variable names than `v` or `acc` in my code ;) )